### PR TITLE
Fix Anvil attack preview and logs

### PR DESCRIPTION
**Pull request by Screaper91**
- Fixes attack preview and log if there is not a hermit opposite the user's active row
- Fixes Anvil returning to player's hand if used by itself and did not deal damage
- Adds log for Anvil being used without dealing damage

### CODE REVIEW

**Comment by Screaper91** [comment]
New PR created since Hyperspace fix may require large refactors before #809 can be merged

**Comment by Lunarmagpie** [comment]
Thanks for opening a new PR. The code looks good after a visual inspection so I will merge this once I verify it works.

File: common/cards/alter-egos/single-use/anvil.ts [hunk 1]
```diff
@@ -32,9 +32,14 @@ const Anvil: SingleUse = {
 		'Do 30hp damage to the Hermit directly opposite your active Hermit on the game board, and 10hp damage to each Hermit below it.',
 	hasAttack: true,
 	attackPreview: (game) => {
-		const targetAmount = getTargetHermits(game, game.currentPlayer).length - 1
-		if (targetAmount === 0) return '$A30$'
-		return `$A30$ + $A10$ x ${targetAmount}`
+		const targets = getTargetHermits(game, game.currentPlayer)
+		if (targets.length === 0) return '0'
```
Comment (Lunarmagpie) [inline common/cards/alter-egos/single-use/anvil.ts:36]: @alicetouhou Should this be '$A0$' or '0'

**Comment by Lunarmagpie** [comment]
Looks good, I just requested @alicetouhou to verify if the color of the 0 should be gray or red. Once he gets back to me I can merge this.

File: common/cards/alter-egos/single-use/anvil.ts (hunk 1 above)
Comment (alicetouhou) [reply common/cards/alter-egos/single-use/anvil.ts:36]: $A0$ as it's a damage value. Thanks for asking!

**Pull request merged**